src/cata_tiles.cpp

```diff
@@ -1317,8 +1317,13 @@ bool cata_tiles::draw_from_id_string(std::string id, TILE_CATEGORY category,
             seed = g->mon_at( tripoint( x, y, z_coord ) );
             break;
     }
+    // this should preserve the deterministic nature of the game based on original seed
+    // even when a tileset changes
+    auto temp = rand();
     srand( seed );
     unsigned int loc_rand = rand();
+    srand( temp );
```

Comment:
I have tested it. The RNG does not go back to its previous state (of course), but it is deterministic.

Determinism is not an immediate requirement to fix this bug, though. I would consider this bug to be much higher priority than an introduced bug in --seed behavior.